### PR TITLE
Search profile

### PR DESCRIPTION
**Pull request by deciocandido**
**FE search/filter profile**

**User Story**
As a user of Cuddly I want to be able to search/filter the profile on in the app

Screenshot 2021-05-17 at 10 30 12

### CODE REVIEW

File: frontend/src/components/navigation/Navbar.jsx [hunk 1]
```diff
@@ -24,6 +25,8 @@ export default function Navbar({ onLogout }) {
           <img className="user nav-icon" src={userURL} alt="Profile" />
         </Link>
 
+        <SearchBar />
+
```
Comment (NeuralAlchemist) [inline frontend/src/components/navigation/Navbar.jsx]: extra lines on line 27 and 29 could be removed

File: frontend/src/components/navigation/Navbar.jsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
 // NPM Packages
 import React from "react";
 import { Link } from "react-router-dom";
+import SearchBar from "../search/SearchBar";
```
Comment (eeels22) [inline frontend/src/components/navigation/Navbar.jsx:6]: Pls could you add an empty line after line 3 and then add `// Project files` above the SearchBar import?

File: frontend/src/components/search/SearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// NPM Packages
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import UserApi from "../../api/UserApi";
```
Comment (eeels22) [inline frontend/src/components/search/SearchBar.jsx:6]: Pls could you add an empty line after line 3 and then add `// Project files` above the `UserApi` import?

File: frontend/src/components/search/SearchCard.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+export default function SearchCard({ match }){
+    // Global state
+    const profile = useRecoilValue(allUsers);
+
+    // Constants
+    const query = match.params.query.toUpperCase();
+    const filteredResults = profile.filter((item) =>
+        item.sender.toUpperCase().match(query)
+    );
+}
```
Comment (NeuralAlchemist) [inline frontend/src/components/search/SearchCard.jsx]: Is this file being used? if not then please remove .

File: frontend/src/components/search/SearchCard.jsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import Profile from "../../components/profile/Profile";
+
+export default function SearchCard({ match }){
+    // Global state
+    const profile = useRecoilValue(allUsers);
+
+    // Constants
+    const query = match.params.query.toUpperCase();
+    const filteredResults = profile.filter((item) =>
+        item.sender.toUpperCase().match(query)
```
Comment (eeels22) [inline frontend/src/components/search/SearchCard.jsx:17]: what is `item.sender` here? should it be `item.name`?

File: frontend/src/components/search/SearchCard.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import { Link } from "react-router-dom";
+
+// Project files
+import { allUsers } from "../../state/usersData";
+import UserApi from "../../api/UserApi";
+import Profile from "../../components/profile/Profile";
+
+export default function SearchCard({ match }){
+    // Global state
+    const profile = useRecoilValue(allUsers);
```
Comment (eeels22) [inline frontend/src/components/search/SearchCard.jsx:12]: Please could we name this `profiles` (plural) so it's clearer it is all the users

File: frontend/src/components/search/SearchBar.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      })
+      .catch((err) => console.error(err));
+    }, [setUsers]);
+
+    // Methods
+    function searchUser(event) {
+        event.preventDefault();
+        //Find out the user by the Id
+        const userId = users.filter((user) => user.name.toUpperCase().match(query.toUpperCase()));
+        history.push(`/profile/${userId[0].id}`);
```
Comment (eeels22) [inline frontend/src/components/search/SearchBar.jsx]: Could we add some error handling if the profile doesn't exist and show an alert? 
Screenshot 2021-05-17 at 10 58 33

Comment (NeuralAlchemist) [reply frontend/src/components/search/SearchBar.jsx]: a suggestion, before history.push you can have if-else and check that userId is not empty, if it is do alert ("no such user found")

**Review by NeuralAlchemist — APPROVED** [review]
Nice feature work!

File: frontend/src/components/search/SearchBar.jsx (hunk 7 above)
Comment (deciocandido) [reply frontend/src/components/search/SearchBar.jsx]: The error has been handled with alert

**Review by eeels22 — APPROVED** [review]
Thanks Decio, looks great!

**Pull request merged**